### ISSUE TITLE
"Filter DNS" works properly only with Wi-Fi

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#191]
Tickets:
QKY-981-92380
WWV-434-11656
LSL-122-19927

**Comment by ameshkov** [comment]
Won't fix because this is a unique issue we've seen only once (all three tickets are from the same user).

**Comment by ameshkov** [comment]
New log is rather weird:
SYW-243-18551

We should look into this one more time.

**Comment by ameshkov** [comment]
Here is what's weird: we use DNS server got from Wi-Fi settings while in fact we are on mobile network now.

**Comment by ameshkov** [comment]
Now it works for mobile too.
